bootstrap4/.js/src/main/scala/io/udash/bootstrap/button/UdashButton.scala

```diff
@@ -1,57 +1,78 @@
 package io.udash.bootstrap
 package button
 
+import com.avsystem.commons.Opt
 import com.avsystem.commons.misc.{AbstractCase, AbstractValueEnum, AbstractValueEnumCompanion, EnumCtx}
 import io.udash._
 import io.udash.bindings.modifiers.Binding
-import io.udash.bootstrap.button.UdashButton.{ButtonClickEvent, UdashButtonJQuery}
+import io.udash.bootstrap.button.UdashButton.{ButtonClickEvent, ButtonTag, UdashButtonJQuery}
 import io.udash.bootstrap.utils._
 import io.udash.wrappers.jquery.{JQuery, jQ}
 import org.scalajs.dom
 import org.scalajs.dom._
 import scalatags.JsDom
-import scalatags.JsDom.TypedTag
 import scalatags.JsDom.all._
 
 import scala.scalajs.js
 
+/**
+ * Options for `UdashButton` component
+ * @param color     A button style, one of the standard bootstrap colors `BootstrapStyles.Color`.
+ * @param size            A button size, one of the standard bootstrap sizes `BootstrapStyles.Size`.
+ * @param outline         If true, selects the outline style for the button. More: <a href="http://getbootstrap.com/docs/4.1/components/buttons/#outline-buttons">Bootstrap Docs</a>.
+ * @param block           If true, rendered button will be a full-width block.
+ * @param tag             HTML tag used in button, one of tags defined in `ButtonTag`
+ * @param href            Href attribute value, used only in case of `ButtonTag.Anchor`
+ * @param customModifiers Sequence of custom modifiers.
+ */
+final case class UdashButtonOptions(
+  color: BootstrapStyles.Color = BootstrapStyles.Color.Primary,
+  size: Opt[BootstrapStyles.Size] = Opt.empty,
+  outline: Boolean = false,
+  block: Boolean = false,
+  tag: ButtonTag = ButtonTag.Button,
+  href: Opt[String] = Opt.Empty,
```

Comment:
If those 2 are used in conjunction, then `ButtonTag.Anchor` should be a class with a `href` instead.